tests/validation/testutils/testutils.go

```diff
@@ -0,0 +1,229 @@
+package validation
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"golang.org/x/crypto/ssh"
+	"io/ioutil"
+	"log"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+type Node struct {
+	Name       string
+	Status     string
+	Roles      string
+	InternalIP string
+	ExternalIP string
+}
+
+type Pod struct {
+	NameSpace string
+	Name      string
+	Ready     string
+	Status    string
+	Restarts  string
+	NodeIP    string
+	Node      string
+}
+
+var config *ssh.ClientConfig
+var SSHKEY string
+var SSHUSER string
+var err error
+
+func checkError(e error) {
+	if e != nil {
+		log.Fatal(err)
+		panic(e)
+	}
+}
+
+func publicKey(path string) ssh.AuthMethod {
+	key, err := ioutil.ReadFile(path)
+	if err != nil {
+		panic(err)
+	}
+	signer, err := ssh.ParsePrivateKey(key)
+	if err != nil {
+		panic(err)
+	}
+	return ssh.PublicKeys(signer)
+}
+
+func ConfigureSSH(host string, SSHUser string, SSHKey string) *ssh.Client {
+	config = &ssh.ClientConfig{
+		User: SSHUser,
+		Auth: []ssh.AuthMethod{
+			publicKey(SSHKey),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	conn, err := ssh.Dial("tcp", host, config)
+	checkError(err)
+	return conn
+}
+
+func runsshCommand(cmd string, conn *ssh.Client) string  {
+	session, err := conn.NewSession()
+	if err != nil {
+		panic(err)
+	}
+	defer session.Close()
+	var stdoutBuf bytes.Buffer
+	var stderrBuf bytes.Buffer
+	session.Stdout = &stdoutBuf
+	session.Stderr = &stderrBuf
+
+	if err := session.Run(cmd); err != nil {
+		log.Println(session.Stdout)
+		log.Fatal("Error on command execution", err.Error())
+	}
+	return fmt.Sprintf("%s", stdoutBuf.String())
+}
+
+func RunCmdOnNode(cmd string, ServerIP string, SSHUser string, SSHKey string) string {
```

Comment:
Can you add basic comments on what this and other exported functions in this file do? I feel that if we need to extend nightly testing it would be helpful for others.